src/components/Pagination/Pagination.tsx

```diff
@@ -1 +1,91 @@
-export const Pagination = () => {};
+import React, { memo } from 'react';
+import cn from 'classnames';
+import { getNumbers } from '../../utils';
+
+type Props = {
+  total: number,
+  perPage: number,
+  currentPage: number,
+  onPageChange: (page: number) => void,
+};
+
+export const Pagination: React.FC<Props> = memo(({
+  perPage,
+  total,
+  currentPage,
+  onPageChange,
+}) => {
+  const lastPage = Math.ceil(total / perPage);
+  const allPages = getNumbers(1, lastPage);
+
+  const handleClickPrevPage = () => {
+    onPageChange(
+      currentPage > 1
+        ? currentPage - 1
+        : currentPage,
+    );
+  };
+
+  const handleClickNextPage = () => {
+    onPageChange(
+      currentPage < lastPage
+        ? currentPage + 1
+        : currentPage,
+    );
+  };
+
+  return (
+    <ul className="pagination">
+      <li
+        className={cn(
+          'page-item',
+          { disabled: currentPage === 1 },
+        )}
+      >
+        <a
+          data-cy="prevLink"
+          className="page-link"
+          href="#prev"
+          aria-disabled={currentPage === 1}
+          onClick={handleClickPrevPage}
+        >
+          «
+        </a>
+      </li>
+      {allPages.map(page => (
+        <li
+          key={page}
+          className={cn(
+            'page-item',
+            { active: currentPage === page },
+          )}
+        >
+          <a
+            data-cy="pageLink"
+            className="page-link"
+            href={`#${page}`}
+            onClick={() => onPageChange(page)}
+          >
```

Comment:
Create separate handler too

`const onPageChangeHanler = (page: number) => () => onPageChange(page)`

```suggestion
            onClick={onPageChangeHanler(page)}
          >
```

Reply:
Fixed everything, thanks!